src/commands/index-command.php

```diff
@@ -156,11 +156,21 @@ public function index( $args = null, $assoc_args = null ) {
 	 * @return void
 	 */
 	protected function run_indexation_actions( $assoc_args ) {
+		// See if we need to clear all indexables before repopulating.
 		if ( isset( $assoc_args['reindex'] ) ) {
+			
+			// Argument --skip-confirmation to prevent confirmation (for automated systems).
 			if ( ! isset( $assoc_args['skip-confirmation'] ) ) {
 				WP_CLI::confirm( 'This will clear all previously indexed objects. Are you certain you wish to proceed?' );
 			}
+
+			// Truncate the tables.
 			$this->clear();
+
+			// Delete the transients to make sure re-indexing runs every time.
+			\delete_transient( $this->post_indexation_action::TRANSIENT_CACHE_KEY );
```

Comment:
I would call the class constant this way, the same for the two lines below
```suggestion
			\delete_transient( Indexable_Post_Indexation_Action::TRANSIENT_CACHE_KEY );
```

Reply:
Thanks, changed them 👍